core/src/main/java/org/apache/druid/java/util/http/client/pool/ResourcePool.java

```diff
@@ -249,6 +302,7 @@ V get()
         throw new RuntimeException(e);
       }
 
+      numLentResources++;
```

Comment:
Please move it to right after you take an item out of resourceHolderList. That will instil more confidence that this variable is not out of sync with resourceHolderList.

Reply:
numLentResources is not a direct substitute for maxSize - resourceHolderList.size().
numLentResources + deficit contributes to that number. So it is intended to increment it only after the change in deficit has been determined

Reply:
makes sense. thanks for clarifying.